piano-hero/core/src/main/java/com/novoda/pianohero/Brain.java

```diff
@@ -0,0 +1,73 @@
+package com.novoda.pianohero;
+
+public class Brain {
+
+    private static final Callback NOOP_CALLBACK = new Callback() {
+        @Override
+        public void onSequenceComplete() {
+            // no op
+        }
+    };
+
+    private final OnSequenceUpdatedCallback onSequenceUpdatedCallback;
+    private Callback callback = NOOP_CALLBACK;
+
+    private Sequence sequence;
+
+    Brain(OnSequenceUpdatedCallback onSequenceUpdatedCallback) {
+        this.onSequenceUpdatedCallback = onSequenceUpdatedCallback;
+    }
+
+    public void attach(Callback callback) {
+        this.callback = callback;
+    }
+
+    public void removeCallbacks() {
+        this.callback = NOOP_CALLBACK;
+    }
+
+    public void start(Sequence sequence) {
+        this.sequence = sequence;
+        onSequenceUpdatedCallback.onNext(sequence);
+    }
+
+    public void onNotesPlayed(Note... notes) {
```

Comment:
in most cases we will. the soft keyboard input only allows for one note at a time, so this was easier + for tests.

this is just sugar around the method below which passes notes